pkg/controllers/cliconfig/cli_config_controller.go

```diff
@@ -0,0 +1,472 @@
+)
+
+func init() {
+	utilruntime.Must(osinv1.Install(scheme))
+}
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"OAuthConfigDegraded",
+	"OAuthSessionSecretDegraded",
+	"OAuthServerIngressDegraded",
+	"OAuthServerServiceDegraded",
+	"AuthCLIConfigProgressing",
+)
+
+type cliConfigController struct {
+	services corev1lister.ServiceLister
+	route    routev1lister.RouteLister
+
+	auth           operatorv1client.AuthenticationsGetter
+	configMaps     corev1client.ConfigMapsGetter
+	secrets        corev1client.SecretsGetter
+	operatorClient v1helpers.OperatorClient
+}
+
+func NewCLIConfigController(kubeInformersForTargetNamespace informers.SharedInformerFactory, secrets corev1client.SecretsGetter, configMaps corev1client.ConfigMapsGetter,
+	operatorClient v1helpers.OperatorClient, authentication operatorv1client.AuthenticationsGetter, routeInformer routeinformer.RouteInformer, recorder events.Recorder) factory.Controller {
+	c := &cliConfigController{
+		services:       kubeInformersForTargetNamespace.Core().V1().Services().Lister(),
+		route:          routeInformer.Lister(),
+		secrets:        secrets,
+		configMaps:     configMaps,
+		operatorClient: operatorClient,
+		auth:           authentication,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Informer(),
+		routeInformer.Informer(),
+		operatorClient.Informer(),
+	).ResyncEvery(30*time.Second).WithSync(c.sync).ToController("CLIConfig", recorder.WithComponentSuffix("cli-config-controller"))
+}
+
+func (c *cliConfigController) getRoute() (*routev1.Route, []operatorv1.OperatorCondition) {
+	// route is a pre-requirement for this sync
+	// it is created by metadata controller
+	// if route does not exists, do nothing and wait
+	route, err := c.route.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return nil, []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthServerIngressDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "GetFailed",
+				Message: fmt.Sprintf("Unable to get oauth server route: %v", err),
+			},
+		}
+	}
+	return route, nil
+}
+
+func (c *cliConfigController) getService() (*corev1.Service, []operatorv1.OperatorCondition) {
+	// service is a pre-requirement for this sync
+	// it is created by serviceca controller
+	// if service does not exists, do nothing and wait
+	service, err := c.services.Services("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return nil, []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthServerServiceDegraded",
```

Comment:
suddenly you have two conditions that appear very similar: `OAuthServiceDegraded` and `OAuthServerServiceDegraded`. Why don't you keep one (preferrably at the controller that sets the service) and have the other controllers just log a failure/use a generic controller-specific condition as shown in the other comment?

Reply:
i went with OAuthConfigServiceDegraded